### PR TITLE
Dev lodf

### LINKED ISSUE #49: Implement LODF and OTDF calculations for contingency analysis.

### PR DESCRIPTION
**Pull request by sourabhdalvi**
Added LODF Calculation and Test
Fixes #49

### CODE REVIEW

File: src/utils/lodf_calculations.jl [hunk 1]
```diff
@@ -0,0 +1,11 @@
+function buildlodf(branches::Array{T}, nodes::Array{Bus}, dist_slack::Array{Float64}=[0.1] ) where {T<:Branch}
+    linecount = length(branches)
+    ptdf , a = PowerSystems.buildptdf(branches,nodes,dist_slack)
+    H = zeros(Float64,linecount,linecount)
+    H = gemm('N','N',ptdf,a)
+    h = diag(H)
+    h = reshape(h,(linecount,1))
+    LODF = H ./ (ones(Float64,linecount,linecount) - gemm('N','T',ones(Float64,linecount,1),h))
```
Comment (jd-lara) [inline src/utils/lodf_calculations.jl]: Why there are two operations on `h`? seems inefficient to allocate it twice. The same applies to the element-wise division

Comment (sourabhdalvi) [reply src/utils/lodf_calculations.jl]: '''
    H = gemm('N','N',ptdf,a)
    h = reshape(diag(gemm('N','N',ptdf,a)),(linecount,1))
    LODF = gemm('N','N',ptdf,a) ./ (ones(Float64,linecount,linecount) - gemm('N','T',ones(Float64,linecount,1),h))
    LODF = LODF - Array(Diagonal(LODF)) - eye(linecount,linecount) 
'''
Is this what you want ?

Comment (jd-lara) [reply src/utils/lodf_calculations.jl]: Yes and also why having the element-wise operation `H ./ (ones(Float64,linecount,linecount)` ?

Comment (sourabhdalvi) [reply src/utils/lodf_calculations.jl]:
```
    H = gemm('N','N',ptdf,a)
    h = reshape(diag(gemm('N','N',ptdf,a)),(linecount,1))
    (Dem, dipiv, dinfo) = getrf!(eye(Float64,linecount,linecount) - Array(Diagonal(H)))
    LODF = gemm('N','N',H,getri!(Dem,dipiv))
    LODF = LODF - Array(Diagonal(LODF)) - eye(linecount,linecount) 
```
This works faster

**Comment by jd-lara** [comment]
@sourabhdalvi the new changes looks good, you can merge the branch.

**Pull request merged**